### PR TITLE
Move helm-chart-released-already check into release script.

### PR DESCRIPTION
**Pull request by danielhoherd**
## Description

Move the check for an already published helm chart farther down the line so chart version mismatches don't block code testing.

## 🧪  Testing

Manually tested the release process on my laptop with an already published version.

### CODE REVIEW

**Review by sjmiller609 — APPROVED** [review]

**Comment by sjmiller609** [comment]
In my opinion, we should leave the test how it is for the sake of convenience. But if you feel it is more convenient this way, that's fine by me. The way I thought of it is that any code addition is invalid if the version is already published because published versions are immutable except for patch version zero, which we overwrite sometimes. If the current behavior is too annoying because it takes a long time to fail for some small thing, I would think it may be better to move the test earlier in the process so it fails CI faster rather than waiting until the final moment of release to error.

**Comment by jpweber** [comment]
@danielhoherd is your take on this, let it run the tests and then if it trips this alert towards the end it prevents it from going somewhere, but you can at least see if your tests work before having to go figure out the version number stuff?

**Comment by danielhoherd** [comment]
Right. The way it is now, as soon as we do a release, tests on every branch except release branches will fail because they all point to a release that has already occurred.

I think the question can be answered by asking ourselves "Where in the release process do we want to bump the version number?" If the answer is "before doing a release" then my change is relevant. If the answer is "right after doing a release" then the existing tests are OK, but still require everybody to rebase against master after a release and will require whoever does a release to actually make that change.

**Comment by jpweber** [comment]
I'm on the side of leave it to the end. Chart versions in the master branch are pretty irrelevant until a release branch is made. This feels like it fits more as a gate to releasing a helm chart than a gate to pushing code to master/main/features/bugfix branches

**Review by jpweber — APPROVED** [review]

**Pull request merged**
